assets/js/theme/global/maintenanceMode.js

```diff
@@ -7,17 +7,43 @@ import $ from 'jquery';
 export default function (maintenanceMode = {}) {
     const header = maintenanceMode.header;
     const notice = maintenanceMode.notice;
+    const password = maintenanceMode.password;
+    const securePath = maintenanceMode.secure_path || '';
 
-    if (!(header && notice)) {
+    // Return if header & notice is null or if securePath is not defined (means inside theme editor)
+    if (!(header && notice) || !securePath) {
         return;
     }
 
-    const $element = $('<div>', {
-        id: 'maintenance-notice',
-        class: 'maintenanceNotice',
-    });
+    if (password) {
+        const $element = $('<div>', {
+            class: 'adminBar',
+        });
 
-    $element.html(`<p class="maintenanceNotice-header">${header}</p>${notice}`);
+        const url = encodeURIComponent(window.location.pathname + window.location.search);
```

Comment:
hmmm...  i think `window.location.pathname` might not be supported in IE:  https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/pathname